pkg/util/util.go

```diff
@@ -378,46 +378,27 @@ func getIPFromInterface(intfName string, nw networkInterfacer) (net.IP, error) {
 	return nil, nil
 }
 
-func flagsSet(flags net.Flags, test net.Flags) bool {
-	return flags&test != 0
-}
-
-func flagsClear(flags net.Flags, test net.Flags) bool {
-	return flags&test == 0
-}
-
-func chooseHostInterfaceNativeGo() (net.IP, error) {
-	intfs, err := net.Interfaces()
+//chooseHostInterfaceNativeGo is a method used to fetch an IP for a daemon.
+//It iterates over the interfaces returned by Interfaces() method
+//and picks the first interface which is not Loopback.
+//This method may pick docker or other bridges
+func chooseHostInterfaceNativeGo(nw networkInterfacer) (net.IP, error) {
```

Comment:
This was for supporting non Linux based systems , like mac and maybe windows.
should 127.0.0.1 be default if there is no other interface on the device ?I thought not.(you have replied it wont work in another PR)
As of now , if a developer is is using his system without any internet connection he cannot start proxy and apiserver .So he has to have a interface , a fake one atleast.